src/App.tsx

```diff
@@ -1,27 +1,79 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+enum Sort {
+  ALL = 'all',
+  FIVEFIRST = 'fivefirst',
+  REDGOODS = 'redgoods',
+}
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [modGood, setModGood] = useState<Sort | undefined>(undefined);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  useEffect(() => {
+    const funcGoodsInfo = async () => {
+      let result;
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      switch (modGood) {
+        case Sort.ALL:
+          result = await getAll().then(allGoods => setGoods(allGoods));
+          break;
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+        case Sort.FIVEFIRST:
+          result = await get5First().then(items => setGoods(items
+            .sort((item1, item2) => item1.name.localeCompare(item2.name))
+            .slice(0, 5)));
```

Comment:
would be good to move all related logic to `get5First()`. And just result with `setGoods`